openmc/mgxs/mgxs.py

```diff
@@ -857,14 +882,172 @@ def get_subdomain_avg_xs(self, subdomains='all'):
+            slice_xs.tallies[tally_type] = tally_slice
+
+        # Assign sliced energy group structure to sliced MGXS
+        if groups:
+            new_group_edges = []
+            for group in groups:
+                group_edges = self.energy_groups.get_group_bounds(group)
+                new_group_edges.extend(group_edges)
+            new_group_edges = np.unique(new_group_edges)
+            slice_xs.energy_groups.group_edges = sorted(new_group_edges)
+
+        # Assign sliced nuclides to sliced MGXS
+        if nuclides:
+            slice_xs.nuclides = nuclides
+
+        slice_xs.sparse = self.sparse
+        return slice_xs
+
+    def can_merge(self, other):
+        """Determine if another MGXS can be merged with this one
+
+        If results have been loaded from a statepoint, then MGXS are only
+        mergeable along one and only one of enegy groups or nuclides.
+
+        Parameters
+        ----------
+        other : MGXS
+            MGXS to check for merging
+
+        """
+
+        if not isinstance(other, type(self)):
+            return False
+
+        # Compare reaction type, energy groups, nuclides, domain type
+        if self.rxn_type != other.rxn_type:
+            return False
+        elif not self.energy_groups.can_merge(other.energy_groups):
+            return False
+        elif self.by_nuclide != other.by_nuclide:
+            return False
+        elif self.domain_type != other.domain_type:
+            return False
+        elif 'distribcell' not in self.domain_type and self.domain != other.domain:
+            return False
+        elif len(self.tallies) != len(other.tallies):
+            return False
+
+        # See if each individual tally is mergeable
+        for tally_key in self.tallies:
+            if not self.tallies[tally_key].can_merge(other.tallies[tally_key]):
+                return False
+
+        # If all conditionals pass then MGXS are mergeable
+        return True
+
+    def merge(self, other):
+        """Merge another MGXS with this one
+
+        If results have been loaded from a statepoint, then MGXS are only
+        mergeable along one and only one of energy groups or nuclides.
+
+        Parameters
+        ----------
+        other : MGXS
+            MGXS to merge with this one
+
+        Returns
+        -------
+        merged_mgxs : MGXS
+            Merged MGXS
+
+        """
+
+        if not self.can_merge(other):
+            raise ValueError('Unable to merge MGXS')
+
+        # Create deep copy of tally to return as merged tally
+        merged_mgxs = copy.deepcopy(self)
+        merged_mgxs._derived = True
+
+        # Merge energy groups
+        if self.energy_groups != other.energy_groups:
+            merged_groups = self.energy_groups.merge(other.energy_groups)
+            merged_mgxs.energy_groups = merged_groups
+
+        # Merge nuclides
+        if self.nuclides != other.nuclides:
```

Comment:
In order to merge MGXS, it looks like the MGXS objects either must have the same set of nuclides or the sets of nuclide sets must be mutually exclusive. Is that correct?